clang/include/clang/Basic/Attr.td

```diff
@@ -1622,6 +1636,33 @@ def SYCLAddIRAttrCommonMembers : SYCLAddIRAttrMemberCodeHolder<[{
                                       ValueE->getType()->isSignedIntegerType());
       return std::string(IntegerStrBuffer);
     }
+    if (const auto *InitListE = dyn_cast<InitListExpr>(ValueE)) {
+      if (InitListE->isStringLiteralInit()) {
+        const Expr *StringInitE = InitListE->getInit(0)->IgnoreParenImpCasts();
+        return getValidAttributeValueAsString(StringInitE, Context);
+      }
+
+      SmallString<10> StrBuffer;
+      for (const auto *InitE : InitListE->inits()) {
+        const Expr *InitNoImpCastE = InitE->IgnoreParenImpCasts();
+        char C = 0;
+        if (const auto *CharacterVal =
+              dyn_cast<CharacterLiteral>(InitNoImpCastE))
+          C = static_cast<char>(CharacterVal->getValue());
+        else if (const auto *IntegerVal =
+              dyn_cast<IntegerLiteral>(InitNoImpCastE))
+          C = static_cast<char>(IntegerVal->getValue().getZExtValue());
```

Comment:
Since this is supposed to be an integer value representing a character I believe it is safe to assume it is an unsigned value. I don't know if there's a way to pick based on the signedness of `char`?

Reply:
But, can I put say `-2` to an initializer list?

Reply:
You are right, that looks to be allowed, though hopefully it will never be applicable. I have have changed it to use `getSExtValue`.